cpp/FileTransferAgent.hpp

```diff
@@ -27,6 +33,94 @@ class IStorageClient;
 
 class FileTransferExecutionResult;
 
+constexpr unsigned long MILL_SECONDS_IN_SECOND = 1000;
+
+class RetryContext
+{
+  public:
+    RetryContext(std::string &fileName):
+    m_retryCount(0),
+    m_putFileName(fileName),
+    m_maxRetryCount(10),
+    m_minSleepTimeInSecs(3), //3 seconds
+    m_maxSleepTimeInSecs(180), //180 seconds is the max sleep time
+    m_timeoutInSecs(600) // timeout 600 seconds.
+    {
+        m_startTime = (ulong)time(NULL);
+    }
+
+    /**
+     * It is retryable if put file status is failed
+     * And retry count is in the limits
+     * And total elapsed time is less than the timeout value specified.
+     *
+     * @param outcome: Put upload status.
+     * @return whether to retry or not.
+     */
+    bool isRetryable(RemoteStorageRequestOutcome outcome)
+    {
+        bool outcomeStatus = ((outcome != RemoteStorageRequestOutcome::SUCCESS) &&
```

Comment:
This is not very readable. What is outcomeStatus supposed to represent ?

Reply:
I agree to slit out the OUTCOME. Leave this class only to handle retry/sleep strategy.